statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+            Represents which cluster to leave out when computing
+            'delete 1' jackknife replicate weights
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        # get stratum that the cluster belongs in
+        s = self.sfclust[c]
+        nh = self.ncs[s]
+        w = self.weights.copy()
+        # all weights within the stratum are modified
+        w[self.strat == s] *= nh / float(nh - 1)
+        # but if you're within the cluster to be removed, set as 0
+        w[self.sclust == c] = 0
+        return w
+
+    def _bootstrap_weights(self):
+        """
+        Computes bootstrap replicate weight
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        w = self.weights.copy()
+        clust_count = np.zeros(self.nclust)
+        for s in range(self.nstrat):
+            # how to handle strata w/ only one cluster?
+            w[self.strat == s] *= float(self.ncs[s] - 1) \
+                                         / self.ncs[s]
+            # If there is only one cluster then weights wont change
+            if self.ncs[s] == 1:
+                continue
+
+            # resample array of clusters
+            ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                                       minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, bsn):
+        """
+        Computes mean bootstrap replicate weight
+
+        Parameters
+        ----------
+        bsn : integer
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.nstrat):
+                w[self.strat == s] *= ((float(self.ncs[s] - 1) /
+                                        self.ncs[s])**(1/bsn))
+                # If there is only one or two clusters then weights wont change
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # resample array of clusters in strata s
+                ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                           minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count /= bsn
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
```

Comment:
the a SurveyDesign instance is not a `__init__` argument to work with delegation.
Should SurveyStat now subclass `object`?